include/bout/mesh.hxx

```diff
@@ -691,7 +720,7 @@ class Mesh {
   
   GridDataSource *source; ///< Source for grid data
   
-  Coordinates *coords;    ///< Coordinate system. Initialised to Null
+  Coordinates *coords_centre, *coords_xlow, *coords_ylow, *coords_zlow;    ///< Coordinate system. Initialised to Null
```

Comment:
Will we ever want other locations? If so we might want to use a `map` that takes the location enum to return a coordinates instance, that should then be easier to extend without having to make changes in multiple places. I'm not sure if it would impact performance.

Reply:
:+1: I think a `map` is a good idea